csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -43,11 +43,17 @@ internal static class NativeMethods
         #region InferenceSession API
 
         [DllImport(nativeLib, CharSet = charSet)]
-        public static extern IntPtr /* OrtStatus* */OrtCreateSession(
+        public static extern IntPtr /* OrtStatus* */OrtCreateSessionW(
                                                         IntPtr /* (OrtEnv*) */ environment,
                                                         [MarshalAs(UnmanagedType.LPWStr)]string modelPath, //the model path is consumed as a wchar* in the C-api
                                                         IntPtr /* (OrtSessionOptions*) */sessopnOptions,
                                                         out IntPtr /**/ session);
+        [DllImport(nativeLib, CharSet = charSet)]
+        public static extern IntPtr /* OrtStatus* */OrtCreateSession(
+                                                IntPtr /* (OrtEnv*) */ environment,
+                                                [MarshalAs(UnmanagedType.LPStr)]string modelPath, //the model path is consumed as a wchar* in the C-api
```

Comment:
char* or wchar*?

Reply:
Good point -- the fix is to pass from C# as a byte[] ....this works for both Linux and Windows. Just have to ensure that the byte[] is extracted using the correct encoding -- Unicode or else UTF8, depending on the platform.